### PR TITLE
Use `sets` instead of `for` statement in "IsValidAuthorizationMode"

### PR DESCRIPTION
**Pull request by miaoyq**
**What this PR does / why we need it**:
Use `sets` instead of `for` statement in "IsValidAuthorizationMode", that will make the code more clearly and concisely.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: pkg/kubeapiserver/authorizer/modes/modes.go [hunk 1]
```diff
@@ -29,10 +31,5 @@ var AuthorizationModeChoices = []string{ModeAlwaysAllow, ModeAlwaysDeny, ModeABA
 
 // IsValidAuthorizationMode returns true if the given authorization mode is a valid one for the apiserver
 func IsValidAuthorizationMode(authzMode string) bool {
-	for _, validMode := range AuthorizationModeChoices {
-		if authzMode == validMode {
-			return true
-		}
-	}
-	return false
+	return sets.NewString(AuthorizationModeChoices...).Has(authzMode)
```
Comment (sttts) [inline pkg/kubeapiserver/authorizer/modes/modes.go]: This is much heavier than the simple loop. Don't think it's a good idea.

Comment (liggitt) [reply pkg/kubeapiserver/authorizer/modes/modes.go]: Agree, but this is not called frequently enough to matter either way

Comment (miaoyq) [reply pkg/kubeapiserver/authorizer/modes/modes.go]: @sttts @liggitt Thanks for review, I think it has better readability for using `sets`.

Comment (miaoyq) [reply pkg/kubeapiserver/authorizer/modes/modes.go]: @sttts @liggitt  Thanks for review, I think it has better readability for using `sets`.

**Pull request merged**